src/v2_services/memory.py

```
@@ -0,0 +1,184 @@
+    → contextualize_question()으로 질문 재구성 (대화 기록이 있을 때만)
+    → chat_service.py에서 재구성된 질문으로 RAG 파이프라인 실행
+    → save_to_history()로 이번 질문/답변을 기록에 저장
+
+[메모리 저장 방식]
+현재는 Python 딕셔너리(인메모리) 방식입니다.
+- 장점: 추가 인프라 없이 바로 사용 가능
+- 단점: 서버를 재시작하면 모든 대화 기록이 초기화됨
+- 향후: DB(PostgreSQL)에 저장하도록 이 파일만 수정하면 됨
+"""
+
+from typing import Dict
```

Ruff (0.15.5)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)